### DISCUSSION TITLE
Supabase client cause reload on windows focus using Next.js

### DISCUSSION THREAD

**Discussion by rmrbytes** [supabase/supabase#11506] [General]
Hello:

I am using Next.js with supabase and the observation is regards the reload when the page goes out of focus and returns back (say moving from one tab on a browser and returning back to the page)

I notice that Next.js pages that have the `const supabase = useSupabaseClient()` code re-loads the page while pages that **do not require the supabase object does not**.  For example:

```
import { useSupabaseClient } from '@supabase/auth-helpers-react'

const Page = () => {
  
    console.log('page does load again')
    ....
   const supabase = useSupabaseClient()
     ....
     return (
        ....
   )
}

export default Page
```

Why does this happen and is there a way to avoid it?

The result of this being that the entire page is reloaded including all the API calls etc. And this happens every time the page loses focus and returns back.

Thanks

**Comment by rmrbytes** [comment]
It seems that this behaviour is related to my page code having a  useEffect on user

```
import ...

const Page = () => {
   const supabase = ...
   const user = useUser()
  ....
  
  useEffect(()=> {
      console.log('this gets called')

   }, [user])
  ...
  return(
  ...
  )
}

export default Page

```

**Comment by minotavi11** [comment]
hi, did you manage to solve this issue?

**Comment by rmrbytes** [comment]
Its been a long while @minotavi11 and do not exactly recollect. I think I moved the supabase code to a Context and saved the reload. But not sure if that it was. This was an experimental project to use Supabase.
